osf/migrations/0177_update_schema_data.py

```diff
@@ -0,0 +1,285 @@
+    """
+    Modify a subquestion's qid, title, description, and help text, because
+    these are often absent.
+    - Modify a subquestion's qid to be of the format "parent-id.current-id", to
+      reflect its nested nature, to ensure uniqueness
+    - For the first nested subquestion, transfer the parent's title, description, and help.
+    """
+    property['qid'] = '{}.{}'.format(get_answer_id(question), property.get('id', ''))
+    if not index:
+        title = question.get('title', '')
+        description = question.get('description', '')
+        help = question.get('help', '')
+        if not property.get('title', '') and not property.get('description'):
+            property['title'] = title
+        if not property.get('description', ''):
+            property['description'] = description
+        if not property.get('help', ''):
+            property['help'] = help
+    return property
+
+
+def format_question(state, rs, question, sub=False):
+    """
+    Split the original question from the schema into multiple schema blocks, all of
+    which have the same chunk_id, to link them.
+    """
+    # If there are subquestions, recurse and format subquestions
+    if question.get('properties'):
+        # Creates section or subsection
+        create_block(
+            state,
+            rs.id,
+            block_type='subsection-heading' if sub else 'section-heading',
+            display_text=question.get('title', '') or question.get('description', ''),
+        )
+        for index, property in enumerate(question.get('properties')):
+            format_question(state, rs, format_property(question, property, index), sub=True)
+    else:
+        # All form blocks related to a particular question share the same chunk_id.
+        chunk_id = generate_object_id()
+        title, description, help, example = find_title_description_help_example(rs, question)
+
+        # Creates question title block
+        create_block(
+            state,
+            rs.id,
+            block_type='question-title',
+            display_text=title,
+            help_text='' if description else help,
+            example_text=example,
+            chunk_id=chunk_id
+        )
+
+        # Creates paragraph block (question description)
+        if description:
+            create_block(
+                state,
+                rs.id,
+                block_type='paragraph',
+                display_text=description,
+                help_text=help,
+                chunk_id=chunk_id,
+            )
+
+        # Creates question input block - this block will correspond to an answer
+        # Map the original schema section format to the new block_type, and create a schema block
+        block_type = FORMAT_TYPE_TO_TYPE_MAP[(question.get('format'), question.get('type'))]
+        create_block(
+            state,
+            rs.id,
+            block_type,
+            required=question.get('required', False),
```

Comment:
Only the form block corresponding to the question's answer will possibly have a required=True. Other form blocks corresponding to the same question (question title, paragraph, multiple choice options all have required=False).